qcengine/procedures/torsiondrive.py

```diff
@@ -0,0 +1,227 @@
+                grid_point: [self._spawn_optimization(grid_point, job, input_model, config) for job in jobs]
+                for grid_point, jobs in next_jobs.items()
+            }
+
+            for grid_point, results in grid_point_results.items():
+
+                failed_results = [result for result in results if not result.success]
+
+                if len(failed_results) > 0:
+
+                    error_message = failed_results[0].error.error_message
+                    error = {
+                        "error_type": "unknown",
+                        "error_message": f"TorsionDrive error at {grid_point}:\n{error_message}",
+                    }
+                    break
+
+                optimization_results[grid_point].extend(results)
+
+            if error is not None:
+                break
+
+            task_results = {
+                grid_point: [
+                    (
+                        result.initial_molecule.geometry.flatten().tolist(),
+                        result.final_molecule.geometry.flatten().tolist(),
+                        result.energies[-1],
+                    )
+                    for result in results
+                ]
+                for grid_point, results in grid_point_results.items()
+            }
+
+            torsiondrive.td_api.update_state(state, {**task_results})
+
+        output_data = input_model.dict()
+        output_data["provenance"] = {
+            "creator": "TorsionDrive",
+            "routine": "torsiondrive.td_api.next_jobs_from_state",
+            "version": torsiondrive.__version__,
+        }
+        output_data["success"] = error is None
+
+        if error is None:
+
+            output_data["final_energies"], output_data["final_molecules"] = {}, {}
+
+            for grid_point, results in optimization_results.items():
+
+                final_energy, final_molecule = self._find_final_results(results)
+
+                output_data["final_energies"][grid_point] = final_energy
+                output_data["final_molecules"][grid_point] = final_molecule
+
+            output_data["optimization_history"] = optimization_results
+
+        else:
+            output_data["error"] = error
+
+        return output_data
+
+    def compute(self, input_model: "TorsionDriveInput", config: "TaskConfig") -> "TorsionDriveResult":
+
+        # Capture the stdout and err here to avoid too much nesting in the _compute function
+        with io.StringIO() as stdout:
+            with io.StringIO() as stderr:
+
+                with redirect_stdout(stdout):
+                    with redirect_stderr(stderr):
+
+                        output_data = self._compute(input_model, config)
+
+                output_data["stdout"] = str(stdout.getvalue())
+                output_data["stderr"] = str(stderr.getvalue())
+
+        if "error" not in output_data:
+
+            output_data.pop("schema_name", None)
+            output_data.pop("schema_version", None)
```

Comment:
IIRC at this stage these will have values taken from the parent `input_model` (something like `qcschema_torsion_drive_input`), so we remove them here so that on L152 pydantic will set them to the current `qcschema_torsion_drive_output` values

Reply:
Ah, makes sense! I've added a comment to indicate this